### DISCUSSION TITLE
Facing error to get the data from FIRESTORE using LARAVEL

### DISCUSSION THREAD

**Discussion by AbdullahIftikahr** [kreait/firebase-php#584] [Q&A]
Hi Everyone,
I am working on FIRESTORE with LARAVEL 8. I have integrated the google FIRESTORE service account. I am able to get collection and documents successfully. But when I am trying to get the document snapshot for getting the data it's giving me an error

 **{ "message": "Missing or insufficient permissions.", "code": 7, "status": "PERMISSION_DENIED", "details": [] }**

I am not able to resolve this issue I need community help with it.

I have followed this link for installation.
https://github.com/googleapis/google-cloud-php-firestore#sample
I have also tried this
kreait/firebase-php

Can anyone know how to resolve it?

Thank You.

**Comment by jeromegamez** [comment]
Based on the little information, I can only guess… this could mean that either the Firestore security rules prevent access to the document you're trying to access, or that the Service Account you're using has insufficient permissions.

You could have a look at https://firebase-php.readthedocs.io/en/5.x/troubleshooting.html#forbidden-errors and check if the needed APIs are enabled.

If you're using a service account for authentication, please make sure that you use one that has the needed permissions: https://firebase-php.readthedocs.io/en/5.x/setup.html#google-service-account

Also, check the Firestore Security Rules.

The links above are useful if you're using this SDK or https://github.com/kreait/laravel-firebase

If you're using the google/cloud-firestore library directly, make sure to read the "Authentication" sections of their documentation: https://github.com/googleapis/google-cloud-php#authentication and https://github.com/googleapis/google-cloud-php/blob/master/AUTHENTICATION.md
